packages/store/operators/src/update-item.ts

```diff
@@ -29,10 +36,11 @@ export function updateItem<T>(
     let value: T = null!;
     // Need to check if the new item value will change the existing item value
     // then, only if it will change it then clone the array and set the item
-    if (isStateOperator(operatorOrValue)) {
-      value = operatorOrValue(existing[index] as Readonly<T>);
+    const theOperatorOrValue = operatorOrValue as T | StateOperator<T>;
+    if (isStateOperator(theOperatorOrValue)) {
+      value = theOperatorOrValue(existing[index] as Readonly<T>);
```

Comment:
Typescript will not narrow the type correctly, so I had to assert the type of `operatorOrValue` first.